### ISSUE TITLE
downloading files and showing them in webview

### ISSUE THREAD

**Issue by MarceloEmmerich** [Microsoft/react-native-windows#1832]
We want to use react native UWP to create an app for Windows 10 which displays locally stored html files. We want to download web content, respectively, html, css, images, pdf and video files, save those locally and display them through a react native WebView.

Therefore we used react-native-fs to download these files from the server and saved them to the local pictures folder which is one of the very few folders you can save to. From there we could not bring them to the WebView again because the only working solution was to use the return of require as a source to the WebView eg. <WebView source={{require(“./mysite.html”)}}.../>
This did not work because we could not require anything from the pictures folder.

Basically our approach leads to the following major problems:
we can not require from some path beyond main project path 
we can give html as string to WebView component but it does not load css and images
we can not get write access to the project directory path, where require works 

Is there any known solution other than writing our own download native plugin?

**Comment by rozele** [comment]
@MarceloEmmerich - the `require` call in most cases simply fixes up the URI for a local file. E.g., the something like `require('./foo.html')` will just map to `ms-appx:///ReactAssets/foo.html`.

I think the problem is that `WebView` will only accept URIs of the prefixed with `ms-appdata:///local` or `ms-appdata:///temp`.  The response from `react-native-fs` is a `file://` URI. I think a simple workaround may be as follows:
1. Download the HTML file with `react-native-fs` to either `RNFS.DocumentDirectoryPath + localPathToFile`or `RNFS.TemporaryDirectoryPath + localPathToFile`.
2. Replace the RNFS URI that you load `<WebView/>` with `'ms-appdata:///local/' + localPathToFile` if you download to `RNFS.DocumentDirectoryPath` or to `'ms-appdata:///temp/' + localPathToFile` if you download to `RNFS.TemporaryDirectoryPath`.

Another workaround is that we could patch up the URIs inside the WebView module, but I suspect the above will unblock you for now.

**Comment by rozele** [comment]
Closing due to inactivity.

**Comment by mohamed-sultan** [comment]
@MarceloEmmerich hey how did you finally handle that

**Comment by MarceloEmmerich** [comment]
we rewrote in Xamarin. Honestly, this was not production-ready back then, maybe it is now, haven't checked.
